src/client/actions/user/index.ts

```diff
@@ -328,37 +336,82 @@ const closeCreateUrlModal: () => CloseCreateUrlModalAction = () => ({
   type: CLOSE_CREATE_URL_MODAL,
 })
 
+const urlCreated = (
+  dispatch: ThunkDispatch<
+    GoGovReduxState,
+    void,
+    | CloseCreateUrlModalAction
+    | ResetUserStateAction
+    | SetSuccessMessageAction
+    | SetLastCreatedLinkAction
+  >,
+  shortUrl: string,
+) => {
+  dispatch<void>(getUrlsForUser())
+  dispatch<ResetUserStateAction>(resetUserState())
+  const successMessage = 'Your link has been created'
+  dispatch<SetLastCreatedLinkAction>(setLastCreatedLink(shortUrl))
+  dispatch<SetSuccessMessageAction>(
+    rootActions.setSuccessMessage(successMessage),
+  )
+  dispatch<CloseCreateUrlModalAction>(closeCreateUrlModal())
+}
+
+// API call to create URL
 // If user is not logged in, the createUrl call returns unauthorized,
 // get them to login, else create the url.
-const createUrlOrRedirect = (history: History) => (
-  dispatch: Dispatch<CloseCreateUrlModalAction | SetErrorMessageAction>,
+const createUrlOrRedirect = (history: History) => async (
+  dispatch: ThunkDispatch<
+    GoGovReduxState,
+    void,
+    | CloseCreateUrlModalAction
+    | ResetUserStateAction
+    | SetSuccessMessageAction
+    | SetLastCreatedLinkAction
+    | SetErrorMessageAction
+  >,
   getState: GetReduxState,
 ) => {
-  const { login, user } = getState()
-  if (login.user.id) {
-    return createUrl(dispatch, user)
-      .then(() => {
-        dispatch<CloseCreateUrlModalAction>(closeCreateUrlModal())
-      })
-      .catch((error) => {
-        // Get user to login if the status is unauthorized
-        if (error.name === 401) {
-          history.push(LOGIN_PAGE)
-        } else if (error.message) {
-          dispatch<SetErrorMessageAction>(
-            rootActions.setErrorMessage(error.message),
-          )
-        } else {
-          dispatch<SetErrorMessageAction>(
-            rootActions.setErrorMessage('An unknown error has occurred.'),
-          )
-          console.error(error)
-        }
-      })
+  const { user } = getState()
+  const { shortUrl } = user
+  let { longUrl } = user
+
+  // Test for malformed short URL
+  if (!/^[a-z0-9-]/.test(shortUrl)) {
+    dispatch<SetErrorMessageAction>(
+      rootActions.setErrorMessage(
+        'Short links should only consist of a-z, 0-9 and hyphens.',
+      ),
+    )
+    return
   }
 
-  history.push(LOGIN_PAGE)
-  return null
+  // Append https:// as the protocol is stripped out
+  // TODO: consider using Upgrade-Insecure-Requests header for HTTP
+  if (!/^(http|https):\/\//.test(longUrl)) {
+    longUrl = `https://${longUrl}` // eslint-disable-line no-param-reassign
```

Comment:
There's a [Url library provided by node](https://nodejs.org/api/url.html) that can help to parse links. You can then check the protocol and modify it as you wish. The codebase has some usages that you can refer

Reply:
The URL library's constructor seems to require the protocol in the string passed to the constructor. I get a type error otherwise